### ISSUE TITLE
T2IAdapter for SDXL

### ISSUE THREAD

**Issue by ke1ne** [MoonRide303/Fooocus-MRE#68]
Maybe, https://github.com/lllyasviel/Fooocus/discussions/325

**Comment by barepixels** [comment]
Haha I just asked @lllyasviel  the same thing

T2I-Adapter-SDXL - The smallest and most efficient control models for SDXL!
https://huggingface.co/collections/TencentARC/t2i-adapter-sdxl-64fac9cbf393f30370eeb02f

t2i-adapter-lineart-sdxl-1.0
t2i-adapter-canny-sdxl-1.0
t2i-adapter-depth-zoe-sdxl-1.0
t2i-adapter-depth-midas-sdxl-1.0
t2i-adapter-sketch-sdxl-1.0
t2i-adapter-openpose-sdxl-1.0)

@lllyasviel will you be adding these to Fooocus?

**Comment by barepixels** [comment]
Read more here  
https://www.reddit.com/r/StableDiffusion/comments/16dcyv5/t2iadapter_for_sdxl_just_released_smallest/

**Comment by MoonRide303** [comment]
Loading [T2I-Adapters](https://huggingface.co/blog/t2i-sdxl-adapters) in diffusers format is supported by backend (ComfyUI) since yesterday (commit [1e6b671](https://github.com/comfyanonymous/ComfyUI/commit/1e6b67101cad777319e891afce3c7120e0dc1273)).

Another options are converted [safetensors versions](https://huggingface.co/lllyasviel/sd_control_collection) from [lllyasviel](https://github.com/lllyasviel) (used by his powerful [ControlNet extension](https://github.com/Mikubill/sd-webui-controlnet) for [SD web UI](https://github.com/AUTOMATIC1111/stable-diffusion-webui), recently updated to [support those new models](https://github.com/Mikubill/sd-webui-controlnet/discussions/2039)).

I see the potential in supporting more models (I used LineArt, Sketch and OpenPose in SD 1.5 days with ControlNet 1.1), as well as in T2I approach (smaller VRAM footprint - though Control-LoRAs rank 128 are not bad, either - it's possible to use 2 of them on a simple 8 GB VRAM GPU). Problem is that packing them all into UI could be too much - UI feels a bit overloaded, already, and Fooocus (even MRE) should remain simple and easy to understand UI. I'd also want to have some space for IP-Adapters (when they'll be supported by backend).

So, to sum it up: I am considering it, but I would need to figure out clean way to do so, first (from both UI and implementation point of view).

**Comment by barepixels** [comment]
Currently which XL openpose does Comfyui support?  Kinda tired of the same poses in XL.

**Comment by ikmalsaid** [comment]
Take your time. I'm still using Comfy to try out new stuffs. Hoping for the best of MRE!

**Comment by aimarxjg** [comment]
It can be downloaded on demand. if you need any function, please go to the "author download address" and put it into the corresponding folder to use this function ~ it is convenient to reduce the volume of the compressed package.
